### PR TITLE
add symengine

### PR DESCRIPTION
**Pull request by tjhei**
candidate for 9.1

work in progress (need more testing)

### CODE REVIEW

File: deal.II-toolchain/packages/symengine.package [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+#https://github.com/symengine/symengine/releases/download/v0.4.0/symengine-0.4.0.tar.gz
+
+BUILDCHAIN=cmake
+
+BUILDDIR=${BUILD_PATH}/${EXTRACTSTO}
+INSTALL_PATH=${INSTALL_PATH}/${EXTRACTSTO}
+
+CONFOPTS="-DCMAKE_BUILD_TYPE=Release \
+"
```
Comment (gfcas) [inline deal.II-toolchain/packages/symengine.package:17]: For a minimal installation one could disable the examples and tests
-DBUILD_EXAMPLES:BOOL=OFF \
-DBUILD_TEST:BOOL=OFF \

File: deal.II-toolchain/packages/symengine.package [hunk 2]
```diff
@@ -0,0 +1,30 @@
+VERSION=0.4.0
+CHECKSUM=fd9ae3e98258291ef35b6392faa72ace
```
Comment (gfcas) [inline deal.II-toolchain/packages/symengine.package:2]: We recently installed symengine successfully with version 0.7.0
VERSION=0.7.0
CHECKSUM=72b8f1442fd3084c96e59d9fe87eec1a

**Comment by gfcas** [comment]
At least with dealii-9.2 we recently installed successfully symengine v0.7.0, in the same fashion @tjhei wants to merge here, on a WSL Ubuntu 20.04, plain Ubuntu 20.04 and a linux cluster with Red Hat Enterprise Linux (RHEL) 7.7.

**Comment by gfcas** [comment]
Should we merge this and then I open the update PR for 0.7.0 or should I directly open the 0.7.0 PR?

**Comment by koecher** [comment]
resolved by #198

**Pull request closed without merging**